lib/src/async.dart

```diff
@@ -0,0 +1,185 @@
+import 'dart:async';
+
+import 'package:bloc_helpers/bloc_helpers.dart';
+import 'package:meta/meta.dart';
+import 'package:rxdart/rxdart.dart';
+
+part 'request.dart';
+
+typedef Future<Result> TaskHandler<Parameter, Result>(Parameter parameter);
+
+/// Helper class to implement asynchronous task
+/// Need to implement the [run] method
+abstract class AsyncTaskBloc<Parameter, Result> extends Bloc {
+  final _callPublisher = new PublishSubject<Parameter>();
+
+  // ignore: close_sinks
+  final _resultPublisher = new PublishSubject<Result>();
+
+  // ignore: close_sinks
+  final _runningBehavior = new BehaviorSubject<bool>(seedValue: false);
+
+  AsyncTaskBloc() {
+    onCall.listen(_handleTask);
+  }
+
+  Future<void> _handleTask(Parameter input) async {
+    _runningBehavior.add(true);
+    try {
+      final res = await run(input);
+      _resultPublisher.add(res);
+    } catch (e, s) {
+      _resultPublisher.addError(e, s);
+    } finally {
+      _runningBehavior.add(false);
+    }
+  }
+
+  @mustCallSuper
+  @override
+  FutureOr<void> dispose() async {
```

Comment:
since it's an `async` function the return type can only be a `Future<void>` not `FutureOr<void>`